src/LibraryManager/Manifest.cs

```diff
@@ -228,16 +229,34 @@ private int DeleteLibraryFiles(ILibraryInstallationState state, Action<string> d
         {
             int filesDeleted = 0;
 
-            foreach (string file in state.Files)
+            IProvider provider = _dependencies.GetProvider(state.ProviderId);
+            ILibraryInstallationResult updatedStateResult = provider.UpdateStateAsync(state, new CancellationToken()).Result;
+            if (updatedStateResult.Success)
             {
-                var url = new Uri(file, UriKind.RelativeOrAbsolute);
-
-                if (!url.IsAbsoluteUri)
+                state = updatedStateResult.InstallationState;
+                foreach (string file in state.Files)
+                {
+                    var url = new Uri(file, UriKind.RelativeOrAbsolute);
+
+                    if (!url.IsAbsoluteUri)
+                    {
+                        string relativePath = Path.Combine(state.DestinationPath, file).Replace('\\', '/');
+                        deleteFileAction?.Invoke(relativePath);
+                        filesDeleted++;
+                    }
+                }
+            }
+            else
+            {
+                _hostInteraction.Logger.Log(string.Format(Resources.Text.LibraryInfoUpdateFailed, state.LibraryId), LogLevel.Error);
```

Comment:
Instead of logging the errors here, we could return ILibraryInstallationResult updatedStateResult. I log it at the callers level. The updatedStateResult could have errors for different reasons other than  "Resources.Text.LibraryInfoUpdateFailed".